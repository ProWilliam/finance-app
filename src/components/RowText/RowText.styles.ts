import { StyleSheet } from 'react-native';
import theme from '../../theme/theme';

const StyleRowText = StyleSheet.create({
  rowInfo: {
    marginVertical: 10,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  text: {
    color: theme.light.quintary,
    fontSize:theme.size.fontSmaller
  }
});

export default StyleRowText;