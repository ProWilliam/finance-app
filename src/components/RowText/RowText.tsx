import React from 'react'
import { View, Text } from 'react-native'

// Styles and Types
import styles from './RowText.styles'
import RowTextProps from '../../types/components/RowText.types';


const StyledText: React.FC<RowTextProps> = ({ title, children}) => {
  
  return (
    <View style={styles.rowInfo}>
      <Text testID='title-text' style={styles.text}>{title}</Text>
      <Text testID='content-text' style={styles.text}>{children}</Text>
    </View>
  )
}
export default StyledText;
