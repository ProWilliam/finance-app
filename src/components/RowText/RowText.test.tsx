import React from 'react';
import renderer from 'react-test-renderer';
import RowText from './RowText';

describe('<RowText />', () => {
  it('renders with title and children', () => {
    const props = {
      title: 'Title Text',
      children: 'Content Text',
    };

    const tree = renderer.create(<RowText {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders without crashing with empty title and children', () => {
    const props = {
      title: '',
      children: '',
    };

    const tree = renderer.create(<RowText {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders with dynamic content', () => {
    const props1 = {
      title: 'First Title',
      children: 'First Content',
    };

    const props2 = {
      title: 'Second Title',
      children: 'Second Content',
    };

    let tree = renderer.create(<RowText {...props1} />).toJSON();
    expect(tree).toMatchSnapshot();

    tree = renderer.create(<RowText {...props2} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});