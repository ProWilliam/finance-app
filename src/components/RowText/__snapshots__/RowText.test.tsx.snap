// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RowText /> renders with dynamic content 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginVertical": 10,
    }
  }
>
  <Text
    style={
      {
        "color": "#6a6a6a",
        "fontSize": 14,
      }
    }
    testID="title-text"
  >
    First Title
  </Text>
  <Text
    style={
      {
        "color": "#6a6a6a",
        "fontSize": 14,
      }
    }
    testID="content-text"
  >
    First Content
  </Text>
</View>
`;

exports[`<RowText /> renders with dynamic content 2`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginVertical": 10,
    }
  }
>
  <Text
    style={
      {
        "color": "#6a6a6a",
        "fontSize": 14,
      }
    }
    testID="title-text"
  >
    Second Title
  </Text>
  <Text
    style={
      {
        "color": "#6a6a6a",
        "fontSize": 14,
      }
    }
    testID="content-text"
  >
    Second Content
  </Text>
</View>
`;

exports[`<RowText /> renders with title and children 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginVertical": 10,
    }
  }
>
  <Text
    style={
      {
        "color": "#6a6a6a",
        "fontSize": 14,
      }
    }
    testID="title-text"
  >
    Title Text
  </Text>
  <Text
    style={
      {
        "color": "#6a6a6a",
        "fontSize": 14,
      }
    }
    testID="content-text"
  >
    Content Text
  </Text>
</View>
`;

exports[`<RowText /> renders without crashing with empty title and children 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginVertical": 10,
    }
  }
>
  <Text
    style={
      {
        "color": "#6a6a6a",
        "fontSize": 14,
      }
    }
    testID="title-text"
  />
  <Text
    style={
      {
        "color": "#6a6a6a",
        "fontSize": 14,
      }
    }
    testID="content-text"
  />
</View>
`;
